File: src/algorithms/strings/edit-distance/regex-matching/sources/RegexMatching_test.cpp
/** Correctness tests for the regexMatching function. */
#include "RegexMatching.cpp"
#include <cassert>
#include <iostream>

int main() {
    assert(regexMatching("aab", "c*a*b") == true);
    assert(regexMatching("aa", "a") == false);
    assert(regexMatching("ab", ".*") == true);
    assert(regexMatching("", "") == true);
    assert(regexMatching("aa", "a*") == true);
    assert(regexMatching("aa", ".*") == true);
    assert(regexMatching("aab", "c*a*") == false);
    assert(regexMatching("mississippi", "mis*is*p*.") == false);
    assert(regexMatching("ab", ".*c") == false);
    assert(regexMatching("a", ".") == true);
    assert(regexMatching("b", "a") == false);
    assert(regexMatching("", "a*") == true);
    assert(regexMatching("aaa", "a*a") == true);
    assert(regexMatching("abc", "a.c") == true);
    std::cout << "All tests passed!" << std::endl;
    return 0;
}

File: src/algorithms/strings/edit-distance/regex-matching/RegexMatching_test.java
/** Correctness tests for the RegexMatching algorithm. */
public class RegexMatching_test {
    public static void main(String[] args) {
        assert RegexMatching.regexMatching("aab", "c*a*b") == true;
        assert RegexMatching.regexMatching("aa", "a") == false;
        assert RegexMatching.regexMatching("ab", ".*") == true;
        assert RegexMatching.regexMatching("", "") == true;
        assert RegexMatching.regexMatching("aa", "a*") == true;
        assert RegexMatching.regexMatching("aa", ".*") == true;
        assert RegexMatching.regexMatching("aab", "c*a*") == false;
        assert RegexMatching.regexMatching("mississippi", "mis*is*p*.") == false;
        assert RegexMatching.regexMatching("ab", ".*c") == false;
        assert RegexMatching.regexMatching("a", ".") == true;
        assert RegexMatching.regexMatching("b", "a") == false;
        assert RegexMatching.regexMatching("", "a*") == true;
        assert RegexMatching.regexMatching("aaa", "a*a") == true;
        assert RegexMatching.regexMatching("abc", "a.c") == true;
        System.out.println("All tests passed!");
    }
}
